src/Features/CSharp/Portable/Completion/CompletionProviders/PartialMethodCompletionProvider.cs

```diff
@@ -33,8 +33,17 @@ protected override SyntaxNode GetSyntax(SyntaxToken token)
         protected override int GetTargetCaretPosition(SyntaxNode caretTarget)
         {
             var methodDeclaration = (MethodDeclarationSyntax)caretTarget;
-            var lastStatement = methodDeclaration.Body.Statements.Last();
-            return lastStatement.GetLocation().SourceSpan.End;
+
+            if (methodDeclaration.Body == null)
+            {
+                return methodDeclaration.GetLocation().SourceSpan.End;
+            }
+            else
+            {
+                // move to the end of the last statement in the method
+                var lastStatement = methodDeclaration.Body.Statements.Last();
+                return lastStatement.GetLocation().SourceSpan.End;
+            }
```

Comment:
It looks like this code is same as in OverrideCompletionProvider:
https://github.com/dotnet/roslyn/blob/7fd423dcfacfc0def10f2af07b5bbecab7d51894/src/Features/CSharp/Portable/Completion/CompletionProviders/OverrideCompletionProvider.cs#L196-L206

Consider extracting a helper method to CompletionUtilities.cs such as `GetCaretPositionForDeclaration(SyntaxNode declaration)` by moving the existing `GetTargetCaretPosition` inside `OverrideCompletionProvider` and then calling this helper method from both `OverrideCompletionProvider` and `PartialMethodCompletionProvider`.

Reply:
Yes, I copied from there, and also considered about a helper.

Reply:
agreed with @Neme12 .  Please just create a helper for this.

Reply:
In which class should it be placed? I haven't found a nice place. `CompletionUtilities`?